nomos-services/key-management-system/src/lib.rs

```diff
@@ -12,16 +12,22 @@ use overwatch_rs::services::{ServiceCore, ServiceData, ServiceId};
 use overwatch_rs::DynError;
 use std::fmt::{Debug, Formatter};
 use std::future::Future;
+use std::pin::Pin;
 use tokio::sync::oneshot;
 
 mod backend;
+mod keys;
 mod secure_key;
 
 const KMS_TAG: ServiceId = "KMS";
 
 // TODO: Use [`AsyncFnMut`](https://doc.rust-lang.org/stable/std/ops/trait.AsyncFnMut.html#tymethod.async_call_mut) once it is stabilized.
 pub type KMSOperator = Box<
-    dyn FnMut(&mut dyn SecuredKey) -> Box<dyn Future<Output = Result<(), DynError>>> + Send + Sync,
+    dyn FnMut(
+            &mut Box<dyn SecuredKey + Send + Sync>,
```

Comment:
If you have a `Box<dyn SecuredKey + Send + Sync>` you should be able to use it in a `&mut dyn SecuredKey + Send + Sync`. What was the issue?